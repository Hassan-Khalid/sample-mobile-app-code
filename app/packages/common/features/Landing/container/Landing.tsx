import React from 'react';
import { View } from 'react-native';
import { LoaderComponent } from '../../../../common/components';
import useLandingControl from '../hooks/useLandingControl';
const Landing = () => {
  const { isLoading } = useLandingControl();
  return (
    <View>
      <LoaderComponent loading={isLoading }  />
    </View>
  );
};
Landing.navigationOptions = () => ({
  header: null
});

export default Landing;
