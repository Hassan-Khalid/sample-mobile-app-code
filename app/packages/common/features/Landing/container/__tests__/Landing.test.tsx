import React from 'react';
import { render } from '@testing-library/react-native';
import Landing from '../Landing';

jest.mock('../../hooks/useLandingControl', () => ({
  __esModule: true,
  default: jest.fn(() => ({
    isLoading: true, // or false, depending on your test case
  })),
}));

describe('Landing component', () => {
  it('render component', () => {
    //setup
    const wrapper = render(<Landing />);
    
    // post-conditions
    expect(wrapper.toJSON).toMatchSnapshot();
  });
  
  it('renders LoaderComponent when isLoading is true', () => {
    const { getByTestId } = render(<Landing />);
    const loaderComponent = getByTestId('loader-component');
    
    expect(loaderComponent).toBeTruthy();
  });
  
});
