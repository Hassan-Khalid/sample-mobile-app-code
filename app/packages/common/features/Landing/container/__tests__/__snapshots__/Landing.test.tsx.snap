// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component render component 1`] = `[Function]`;
